types/mapbox-gl/index.d.ts

```diff
@@ -367,12 +367,20 @@ declare namespace mapboxgl {
 
         setPaintProperty(layer: string, name: string, value: any, options?: FilterOptions): this;
 
+        setPaintProperty(layer: string, name: AnyPaintName, value: any, options?: FilterOptions): this;
```

Comment:
This doesn't seem right; isn't `AnyPaintName` a union of string literals? So the first overload will still take precedence?

If you're looking for auto-complete, you'd want to instead make one overload with a parameter `AnyPaintName | (string & {})`.

But as-is, this PR doesn't make sense.